File: app/src/main/java/dev/ragnarok/fenrir/util/InstancesCounter.java
package dev.ragnarok.fenrir.util;

import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.atomic.AtomicInteger;


public class InstancesCounter {

    private final Map<Class<?>, AtomicInteger> map;

    public InstancesCounter() {
        map = new HashMap<>();
    }

    public int incrementAndGet(Class<?> c) {
        AtomicInteger counter = map.get(c);
        if (counter == null) {
            counter = new AtomicInteger();
            map.put(c, counter);
        }

        return counter.incrementAndGet();
    }

    public void fireExists(Class<?> c, int id) {
        AtomicInteger counter = map.get(c);

        if (counter == null) {
            counter = new AtomicInteger();
            map.put(c, counter);
        }

        if (counter.get() < id) {
            counter.set(id);
        }
    }
}

File: app_filegallery/src/main/kotlin/dev/ragnarok/filegallery/util/InstancesCounter.kt
package dev.ragnarok.filegallery.util

import java.util.concurrent.atomic.AtomicLong

class InstancesCounter {
    private val map: MutableMap<Class<*>, AtomicLong>
    fun incrementAndGet(c: Class<*>): Long {
        var counter = map[c]
        if (counter == null) {
            counter = AtomicLong()
            map[c] = counter
        }
        return counter.incrementAndGet()
    }

    fun fireExists(c: Class<*>, id: Long) {
        var counter = map[c]
        if (counter == null) {
            counter = AtomicLong()
            map[c] = counter
        }
        if (counter.get() < id) {
            counter.set(id)
        }
    }

    init {
        map = HashMap()
    }
}
